elastic/kibana
# [NP migration] Shim ILM

## Test plan

- [x] API calls
  - [x] loadNodes
  - [x] loadNodeDetails
  - [x] loadIndexTemplates
  - [x] loadPolicies
  - [x] savePolicy
  - [x] deletePolicy
  - [x] addLifecyclePolicyToTemplate
- [x] Redirects
  - [x] "Cancel" button on Edit/Create policy page goes back to list
- [x] UI metrics
  - [x] UIM_APP_LOAD
  - [x] UIM_POLICY_CREATE
  - [x] UIM_POLICY_UPDATE
  - [x] UIM_POLICY_DELETE
  - [x] UIM_POLICY_ATTACH_INDEX_TEMPLATE
  - [x] UIM_CONFIG_COLD_PHASE
  - [x] UIM_CONFIG_WARM_PHASE
  - [x] UIM_CONFIG_SET_PRIORITY
  - [x] UIM_CONFIG_FREEZE_INDEX
  - [x] UIM_EDIT_CLICK
- [x] Toasts and Fatal errors
- [x] "Learn more" links in editor
- [x] extend_index_management
  - [x] retryLifecycleForIndex API call
  - [x] removeLifecycleForIndex API call
  - [x] addLifecyclePolicyToIndex API call
  - [x] UIM_POLICY_ATTACH_INDEX UI metric
  - [x] UIM_POLICY_DETACH_INDEX UI metric
  - [x] UIM_INDEX_RETRY_STEP UI metric
- [x] Disabling UI with `xpack.ilm.ui.enabled: false`

## Testing the Index Management extensions

1. Create an index lifecycle policy with a Hot Phase that has enabled rollover and max age of a few nanoseconds.
2. Create an index in Console and add the policy to it in Index Management. You should get a modal and then once you've confirmed, a toast.
3. View the index's details to see 
4. Add a document to the index and wait a few seconds
5. Refresh Index Management and you should see a warning callout about errors in the policy
7. Retry the index lifecycle step for the index
8. Remove the index lifecycle policy from the index. You should get a modal and then once you've confirmed, a toast.

## Verification that dependencies still function

### Toasts

![image](https://user-images.githubusercontent.com/1238659/73587504-84e7d300-4471-11ea-8fe0-6bc9bf050328.png)

### Fatal error

![image](https://user-images.githubusercontent.com/1238659/73587502-7ef1f200-4471-11ea-9bc5-4d84bda2dcf4.png)

### Fetching index templates

![image](https://user-images.githubusercontent.com/1238659/73587506-903afe80-4471-11ea-8053-05100c74a845.png)

### Fetching nodes

![image](https://user-images.githubusercontent.com/1238659/73587508-98933980-4471-11ea-9a35-5fea69f66623.png)

### Fetching node details

![image](https://user-images.githubusercontent.com/1238659/73587509-9df08400-4471-11ea-8530-013d426f91b6.png)

### Index Management extension

![image](https://user-images.githubusercontent.com/1238659/73587513-aa74dc80-4471-11ea-9dc4-f1a6295e48ed.png)
